source/include/FreeRTOS_DHCP.h

```diff
@@ -133,6 +125,18 @@
                                                     uint32_t ulIPAddress );
     #endif /* ( ipconfigUSE_DHCP_HOOK != 0 ) */
 
+    #if ( ipconfigDHCP_FALL_BACK_AUTO_IP != 0 )
+        struct xNetworkEndPoint;
+
+/**
+ * @brief When DHCP has failed, the code can assign a Link-Layer
+ *        address, and check if another device already uses the IP-address.
+ *
+ * param[in] pxEndPoint: The end-point that wants to obtain a link-layer address.
+ */
+        void prvPrepareLinkLayerIPLookUp( NetworkEndPoint_t * pxEndPoint );
```

Comment:
Traditionally, "FreeRTOS_Routing.h" and "FreeRTOS_ND.h" are included as the last header files. That means that the earlier header files like "FreeRTOS_DHCP.h" can not refer to typedefs made in "FreeRTOS_Routing.h".

That is the reason that:
~~~diff
-void prvPrepareLinkLayerIPLookUp( NetworkEndPoint_t * pxEndPoint );
+void prvPrepareLinkLayerIPLookUp( struct xNetworkEndPoint * pxEndPoint );
~~~

Reply:
Fixed here: https://github.com/FreeRTOS/FreeRTOS-Plus-TCP/pull/537/commits/75492057e5512ca002cec9bccc458c59522127af